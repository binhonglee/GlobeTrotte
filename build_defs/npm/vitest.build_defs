subinclude("//build_defs/npm")

def vitest_dir(
  name:str,
  srcs:list=[],
  deps:list=[],
  visibility:list=[],
):
  cmd = "zip -r " + name + ".vitest_dir ./"

  return build_rule(
    name = name,
    srcs = srcs,
    cmd = cmd,
    outs = [name + ".vitest_dir"],
    deps = deps,
    test_only = True,
    visibility = visibility
  )

def vitest_tests(
  name:str,
  srcs:list=[],
  vitest_dirs:list=[],
  config:str=CONFIG.VITEST_CONFIG,
  root:str=None,
  deps:list=[],
  visibility:list=[],
  # pnpm complains about  ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL  without first running install
  run_install:bool=False,
):
  test_rules = []

  for src in srcs:
    test = vitest_test(
      name = src[:-8].replace("/", "_"),
      src = src,
      config = config,
      root = root,
      vitest_dirs = vitest_dirs,
      deps = deps,
      visibility = visibility,
      run_install = run_install,
    )
    test_rules += [test]

  return build_rule(
    name = name,
    data = test_rules + srcs,
    test_cmd = "echo 'test'",
    test_only = True,
    visibility = visibility,
    requires = ["npm-test"],
  )

def vitest_test(
  name:str,
  src:str,
  vitest_dirs:list=[],
  config:str=CONFIG.VITEST_CONFIG,
  root:str=None,
  deps:list=[],
  visibility:list=[],
  # pnpm complains about  ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL  without first running install
  run_install:bool=False,
):
  pre_build = _set_node_modules()
  if len(vitest_dirs) > 0:
    pre_build = " && ".join([
      pre_build,
      "; ".join([
        "for z in $(find . -name '*.vitest_dir' -type f)",
        "do unzip -o $z",
        "done",
      ]),
    ])

  if root is not None:
    pre_build += " && cd " + root
  
  if run_install:
    pre_build += " && $TOOLS_NPM i"

  out_dir = _vitest_out_dir(name)
  if CONFIG.CORE_FILES != "":
    deps += [CONFIG.CORE_FILES]

  return build_rule(
    name = name,
    data = [src] + vitest_dirs + deps,
    test_cmd = " && ".join([
      pre_build,
      " ".join([
        _get_npx_tool(),
        "vitest",
        "--coverage",
        "--threads=false",
        "--reporter=junit",
        "--run",
        "$PKG_DIR/" + src,
        "-c",
        "$TOOLS_CONFIG",
      ]),
      "cp vitest_junit test.results",
      "cp vitest_coverage/coverage-final.json test.coverage",
      "mkdir -p " + out_dir,
      "mv vitest_coverage/* " + out_dir,
    ]),
    test_outputs = [out_dir],
    test = True,
    test_only = True,
    test_tools = {
      "config": config,
      "npm": _get_npm_tool(),
    },
    visibility = visibility,
    labels = ['vitest'],
  )

def _vitest_out_dir(name: str):
  return name + "/.nyc_output"

CONFIG.setdefault("VITEST_CONFIG", "vite.config.ts")
