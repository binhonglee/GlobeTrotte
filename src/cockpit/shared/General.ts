import {
  ElNotificationOptions,
  MessageType,
} from "element-ui";
import { WingsStructUtil } from "wings-ts-util";
import HTTPReq from "./HTTPReq";
import User from "@/wings/User";
import Trip from "@/wings/Trip";
import { CombinedVueInstance } from "vue/types/vue";

export default class General {
  public static paramID(
    obj: CombinedVueInstance,
  ): string | undefined {
    /* istanbul ignore next: $route is a pain to mock, using this as a workaround for testing */
    return obj.$route.params.id;
  }

  public static paramNext(
    obj: CombinedVueInstance,
  ): string {
    /* istanbul ignore next: $route is a pain to mock, using this as a workaround for testing */
    try {
      return obj.$route.params.path ?? "";
    } catch (_) {
      return "";
    }
  }

  public static addNext(
    path: string,
    next: string,
  ): string {
    if (next.startsWith("/")) {
      next = next.substr(1, next.length);
    }
    const nexts = next.split("/:");
    const first = nexts.shift()?.split("/").join(".");
    if (nexts.length > 0) {
      next = first + "&" + nexts.join("");
    } else {
      next = first;
    }

    return path + "/:" + next;
  }

  public static getNext(path: string): string {
    if (path !== "") {
      if (!path.startsWith(":")) {
        path = path.split("/:")[1] ?? "";
      } else {
        path = path.substr(1, path.length);
      }
      const paths = path.split("&");
      let next = "";
      while (next === "" || next === undefined) {
        next = paths.shift()?.split(".").join("/");
      }
      path = next + "/:" + paths.join("&");
    }
    return "/" + path;
  }

  public static toNext(obj: CombinedVueInstance): void {
    obj.$router.push(this.getNext(this.paramNext(obj)));
  }

  public static notifConfig(
    title: string,
    message: string,
    type: MessageType,
  ): ElNotificationOptions {
    return {
      message: message,
      title: title,
      type: type,
      duration: 2000,
      offset: 50,
    };
  }

  public static async genUser(id: number): Promise<User> {
    const user = await HTTPReq.genGET("user/" + id);
    return new User(user);
  }

  public static async genTrip(id: number): Promise<Trip> {
    const trip = await HTTPReq.genGET("trip/" + id);
    return new Trip(trip);
  }

  public static getIsCurrentUser(id: number): boolean {
    return this.getCurrentUser().ID === id;
  }

  public static async genCurrentUser(): Promise<User> {
    const id = JSON.parse(await HTTPReq.genGET("whoami"))
      .id;
    if (id === -1) {
      localStorage.clear();
    } else {
      localStorage.setItem(
        "user",
        WingsStructUtil.stringify(await this.genUser(id)),
      );
    }

    return this.getCurrentUser();
  }

  private static getCurrentUser(): User {
    const user = localStorage.getItem("user");
    if (user !== null) {
      return new User(JSON.parse(user));
    }

    return new User();
  }

  public static authSession(): boolean {
    return this.getCurrentUser().ID !== -1;
  }
}
