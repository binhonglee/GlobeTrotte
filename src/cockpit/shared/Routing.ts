import { R } from "@glareshield/all";

import router from "@/router";
import Routes from "@/routes";

export default class Routing extends R {
  protected static siteURI =
    process.env.NODE_ENV === "production"
      ? "https://globetrotte.com"
      : "http://localhost:3000";
  protected static router = router;
  protected static ratelimited = Routes.RateLimited;
  protected static default = Routes.Landing;

  // Unlike genRedirectTo, this is for when you are redirecting back to the same
  // page with different params.The page needs to be refreshed to have the new
  // params dealt with (remounted).
  public static async genRefreshRedirect(
    path: string,
    map: Map<string, string> = new Map<string, string>(),
    id = "",
  ): Promise<void> {
    await this.genRedirectTo(
      Routes.Refresh,
      new Map<string, string>(
        Object.entries({
          next: this.getSubPath(path, map, id)
            .replaceAll("/", ".slash.")
            .replaceAll(":", ".colon.")
            .replaceAll("=", ".equal."),
        }),
      ),
      "",
      true,
    );
  }
}
