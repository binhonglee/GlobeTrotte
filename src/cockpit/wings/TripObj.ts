// DO NOT EDIT THIS FILE DIRECTLY!
// Edits will be overwritten on build.
//
// If you would like to make any changes, please edit the source file (src/wings/struct/trip_obj.wings)
// and run the following command:
// plz build //src/wings/...

import TripBasic from './TripBasic';
import UserBasic from './UserBasic';
import { IWingsStruct } from 'wings-ts-util';

// All information of a single trip.
export default class TripObj implements IWingsStruct {
  [key: string]: any;
  public ID: Number = -1;
  public details: TripBasic = new TripBasic();
  public user: UserBasic = new UserBasic();
  public timeCreated: Date = new Date();
  public lastUpdated: Date = new Date();

  public constructor(obj?: any) {
    if (obj) {
      this.ID = obj.id !== undefined && obj.id !== null ? obj.id : -1;
      this.details = obj.details !== undefined && obj.details !== null ? new TripBasic(obj.details) : new TripBasic();
      this.user = obj.user !== undefined && obj.user !== null ? new UserBasic(obj.user) : new UserBasic();
      this.timeCreated = obj.time_created !== undefined && obj.time_created !== null ? new Date(obj.time_created) : new Date();
      this.lastUpdated = obj.last_updated !== undefined && obj.last_updated !== null ? new Date(obj.last_updated) : new Date();
    }
  }

  public toJsonKey(key: string): string {
    switch (key) {
      case 'ID': {
        return 'id';
      }
      case 'details': {
        return 'details';
      }
      case 'user': {
        return 'user';
      }
      case 'timeCreated': {
        return 'time_created';
      }
      case 'lastUpdated': {
        return 'last_updated';
      }
      default: {
        return key;
      }
    }
  }
}
