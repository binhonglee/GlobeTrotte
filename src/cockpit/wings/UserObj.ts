// DO NOT EDIT THIS FILE DIRECTLY!
// Edits will be overwritten on build.
//
// If you would like to make any changes, please edit the source file (src/wings/struct/user_obj.wings)
// and run the following command:
// plz build //src/wings/...

import TripBasic from './TripBasic';
import { parseArray } from 'wings-ts-util';
import UserBasic from './UserBasic';
import { IWingsStruct } from 'wings-ts-util';

// All information of a single user.
export default class UserObj implements IWingsStruct {
  [key: string]: any;
  public ID: Number = -1;
  public details: UserBasic = new UserBasic();
  public trips: TripBasic[] = [];
  public timeCreated: Date = new Date();

  public constructor(obj?: any) {
    if (obj) {
      this.ID = obj.id !== undefined && obj.id !== null ? obj.id : -1;
      this.details = obj.details !== undefined && obj.details !== null ? new UserBasic(obj.details) : new UserBasic();
      this.trips = obj.trips !== undefined && obj.trips !== null ? parseArray<TripBasic>(TripBasic, obj.trips) : [];
      this.timeCreated = obj.time_created !== undefined && obj.time_created !== null ? new Date(obj.time_created) : new Date();
    }
  }

  public toJsonKey(key: string): string {
    switch (key) {
      case 'ID': {
        return 'id';
      }
      case 'details': {
        return 'details';
      }
      case 'trips': {
        return 'trips';
      }
      case 'timeCreated': {
        return 'time_created';
      }
      default: {
        return key;
      }
    }
  }
}
