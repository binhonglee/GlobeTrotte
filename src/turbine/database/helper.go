package database

// NotExists - Return an array of int from haystack that aren't in the needles
func NotExists(haystack []int, needles []int) []int {
	needs := make(map[int]bool)
	notExists := make(map[int]bool)

	for _, v := range needles {
		needs[v] = true
	}

	for _, v := range haystack {
		if !needs[v] {
			notExists[v] = true
		}
	}
	toReturn := make([]int, len(notExists))
	i := 0
	for key, _ := range notExists {
		toReturn[i] = key
		i++
	}

	return toReturn
}
