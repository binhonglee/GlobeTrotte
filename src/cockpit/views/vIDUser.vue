<template lang="pug">
.get_user
  h1.title {{ user.details.name }}
  .profile_info
    CViewUser(:user="user" :showName="false")
</template>

<script lang="ts">
import { defineComponent } from "vue";
import CViewUser from "@/components/CViewUser.vue";
import General from "@/shared/General";
import UserObj from "@/wings/UserObj";
import Routes from "@/routes";
import Routing from "@/shared/Routing";

interface Data {
  user: UserObj;
}

export default defineComponent({
  components: {
    CViewUser,
  },
  data: (): Data => ({
    user: new UserObj(),
  }),
  async beforeMount(): Promise<void> {
    if (General.paramID() === undefined) {
      await Routing.genRedirectTo(Routes.NotFound);
      return;
    }

    this.$data.user = await General.genUser(Number(General.paramID()));

    if (this.$data.user.ID === -1) {
      await this.$alert("User not found.", "Error", {
        confirmButtonText: "OK",
      });
      await Routing.genRedirectTo(Routes.Landing);
      return;
    }

    if (General.getIsCurrentUser(this.$data.user.ID.valueOf())) {
      await Routing.genRedirectTo(Routes.MyAccount);
    }

    this.$data.user.details.bio = this.$data.user.details.bio.replaceAll(
      "\\\\n",
      "\n",
    );
  },
});
</script>

<style lang="scss" scoped>
@import "../shared/lib";

.profile_info {
  text-align: left;
}
</style>
