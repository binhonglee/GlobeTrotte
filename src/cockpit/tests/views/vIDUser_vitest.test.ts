import vIDUser from "@/views/vIDUser.vue";
import General from "@/shared/General";
import { mountingOptions, wait } from "../helper";
import { alertSpy, stub } from "../vitestSpy";
import { describe, expect, test, vi } from "vitest";
import { mount } from "@vue/test-utils";
import UserObj from "@/wings/UserObj";
import TripObj from "@/wings/TripObj";
import Routing from "@/shared/Routing";
import Routes from "@/routes";

const currentUser = new UserObj({
  id: 10,
  details: {
    id: 10,
    username: "testmockuseraccount",
    name: "MyAccount Test User",
    email: "testmyaccount@globetrotte.com",
  },
  trips: [
    {
      id: 5,
    },
  ],
});

const trip5 = new TripObj({
  id: 5,
  details: {
    id: 5,
    name: "some title",
    description: "",
    private: false,
  },
  time_created: "2020-12-31T00:00:00.888-08:00",
  last_updated: "2020-12-31T00:00:00.888-08:00",
  user: {
    id: 10,
    username: "testmockuseraccount",
    name: "MyAccount Test User",
    email: "testmyaccount@globetrotte.com",
  },
});

describe("vIDUser", () => {
  test("Get User - Has user (self)", async () => {
    const genUser = stub(vi.spyOn(General, "genFromUsername")).resolves(
      currentUser,
    );
    const genTrip = stub(vi.spyOn(General, "genTrip")).resolves(trip5);
    const isSelf = stub(vi.spyOn(General, "getIsCurrentUser")).returns(true);
    const paramID = stub(vi.spyOn(General, "paramID")).returns(
      currentUser.details.username,
    );
    const redirection = stub(
      vi.spyOn(Routing, "genRedirectTo").mockResolvedValue(),
    );
    const wrapper = mount(vIDUser, mountingOptions());
    await wait(0);
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find(".title").text()).toEqual(currentUser.details.name);
    expect(paramID.called()).toBeTruthy();
    expect(isSelf.calledOnce()).toBeTruthy;
    expect(isSelf.args()[0][0]).toEqual(10);
    expect(redirection.calledOnce()).toBeTruthy();
    expect(redirection.args()[0][0]).toEqual(Routes.MyAccount);
    expect(genUser.calledOnce()).toBeTruthy();
    expect(genTrip.calledOnce()).toBeTruthy();
    await genUser.restore();
    await genTrip.restore();
    await isSelf.restore();
    await redirection.restore();
    await paramID.restore();
  });

  test("Get User - Has user (not self)", async () => {
    const genUser = stub(vi.spyOn(General, "genFromUsername")).resolves(
      currentUser,
    );
    const genTrip = stub(vi.spyOn(General, "genTrip")).resolves(trip5);
    const isSelf = stub(vi.spyOn(General, "getIsCurrentUser")).returns(false);
    const paramID = stub(vi.spyOn(General, "paramID")).returns(
      currentUser.details.username,
    );
    const wrapper = mount(vIDUser, mountingOptions());
    await wait(0);
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find(".title").text()).toEqual(currentUser.details.name);
    expect(paramID.called()).toBeTruthy();
    expect(isSelf.calledOnce()).toBeTruthy();
    expect(isSelf.args()[0][0]).toEqual(10);
    expect(genUser.calledOnce()).toBeTruthy();
    expect(genTrip.calledOnce()).toBeTruthy();
    await genUser.restore();
    await genTrip.restore();
    await isSelf.restore();
    await paramID.restore();
  });

  test("Get User - Not found", async () => {
    const genUser = stub(vi.spyOn(General, "genFromUsername")).callsFake(
      async () => {
        // This is so there is enough time for alertSpy to be
        // created before it reaches the alert code
        await wait(0);
        return new UserObj({
          id: -1,
        });
      },
    );
    const paramID = stub(vi.spyOn(General, "paramID")).returns(
      currentUser.details.username,
    );
    const wrapper = mount(vIDUser, mountingOptions());
    const alert = new alertSpy(wrapper);
    await wait(0);
    expect(wrapper.html()).toMatchSnapshot();
    expect(paramID.called()).toBeTruthy();
    expect(genUser.calledOnce()).toBeTruthy();
    expect(alert.calledOnce()).toBeTruthy();
    expect(alert.getTitle()).toEqual("Error");
    expect(alert.getMessage()).toEqual("User not found.");
    expect(alert.getOptions("confirmButtonText")).toEqual("OK");
    await genUser.restore();
    await paramID.restore();
    await alert.restore();
  });

  // Somehow this doesn't work. Not sure why.
  // test("Get User - No Param", async () => {
  //   const paramID = stub(vi.spyOn(General, "paramID")).returns(undefined);
  //   mount(vIDUser, mountingOptions());
  //   await wait(0);
  //   expect(paramID.calledOnce()).toBeTruthy();
  //   await paramID.restore();
  // });
});
