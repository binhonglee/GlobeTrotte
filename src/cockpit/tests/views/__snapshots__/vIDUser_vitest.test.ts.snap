// Vitest Snapshot v1

exports[`vIDUser > Get User - Has user (not self) 1`] = `
<div class="get_user"
     data-v-de4cceec
>
  <div class="head"
       data-v-de4cceec
  >
  </div>
  <h1 class="title"
      data-v-de4cceec
  >
    MyAccount Test User
  </h1>
  <div class="profile_info"
       data-v-de4cceec
  >
    <div class="view_user"
         data-v-311c04db
         data-v-de4cceec
    >
      <div class="userInfo narrow_content"
           data-v-311c04db
      >
        <div class="externalLink"
             data-v-311c04db
        >
        </div>
      </div>
      <div role="separator"
           class="n-divider n-divider--no-title viewUserDivider"
           style="--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-color: rgb(239, 239, 245); --n-text-color: rgb(31, 34, 37); --n-font-weight: 500;"
           data-v-311c04db
      >
        <div class="n-divider__line n-divider__line--left">
        </div>
      </div>
      <div class="viewUserTrips"
           data-v-311c04db
      >
        <h2 data-v-311c04db>
          Trips
        </h2>
        <div class="trip_preview_card wide_preview_card"
             data-v-22a19118
             data-v-311c04db
        >
          <div class="n-card n-card--bordered n-card--hoverable widePreviewCard"
               style="--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-border-radius: 3px; --n-color: #fff; --n-color-modal: #fff; --n-color-popover: #fff; --n-color-target: #18a058; --n-text-color: rgb(51, 54, 57); --n-line-height: 1.6; --n-action-color: rgb(250, 250, 252); --n-title-text-color: rgb(31, 34, 37); --n-title-font-weight: 500; --n-close-color: rgba(122, 122, 122, 1); --n-close-color-hover: rgba(89, 89, 89, 1); --n-close-color-pressed: rgba(149, 149, 149, 1); --n-border-color: rgb(239, 239, 245); --n-box-shadow: 0 1px 2px -2px rgba(0, 0, 0, .08), 0 3px 6px 0 rgba(0, 0, 0, .06), 0 5px 12px 4px rgba(0, 0, 0, .04); --n-padding-top: 19px; --n-padding-bottom: 20px; --n-padding-left: 24px; --n-font-size: 14px; --n-title-font-size: 18px; --n-close-size: 18px;"
               data-v-22a19118
          >
            <div class="n-card__content"
                 style="padding: 0px;"
                 role="none"
            >
              <a class="underline_never color_never tripLink"
                 href="/trip/view/5"
                 rel
                 target
                 data-v-018c0145
                 data-v-22a19118
              >
                <h3 class="tripTitleName"
                    data-v-22a19118
                >
                  some title
                </h3>
                <div role="separator"
                     class="n-divider n-divider--no-title"
                     style="--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-color: rgb(239, 239, 245); --n-text-color: rgb(31, 34, 37); --n-font-weight: 500;"
                     data-v-22a19118
                >
                  <div class="n-divider__line n-divider__line--left">
                  </div>
                </div>
                <div class="tripPreviewInfo"
                     data-v-22a19118
                >
                  <p data-v-22a19118>
                    <a class="underline_hover color_always tripPreviewUserProfileLink"
                       href="/user/10"
                       rel
                       target
                       data-v-018c0145
                       data-v-22a19118
                    >
                      MyAccount Test User
                    </a>
                  </p>
                  <div class="cityTags"
                       data-v-22a19118
                  >
                  </div>
                  <p data-v-22a19118>
                    Last Updated: Dec 31 2020
                  </p>
                </div>
                <div class="endOfCardPadding"
                     data-v-22a19118
                >
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`vIDUser > Get User - Has user (self) 1`] = `
<div class="get_user"
     data-v-de4cceec
>
  <div class="head"
       data-v-de4cceec
  >
  </div>
  <h1 class="title"
      data-v-de4cceec
  >
    MyAccount Test User
  </h1>
  <div class="profile_info"
       data-v-de4cceec
  >
    <div class="view_user"
         data-v-311c04db
         data-v-de4cceec
    >
      <div class="userInfo narrow_content"
           data-v-311c04db
      >
        <div class="externalLink"
             data-v-311c04db
        >
        </div>
      </div>
      <div role="separator"
           class="n-divider n-divider--no-title viewUserDivider"
           style="--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-color: rgb(239, 239, 245); --n-text-color: rgb(31, 34, 37); --n-font-weight: 500;"
           data-v-311c04db
      >
        <div class="n-divider__line n-divider__line--left">
        </div>
      </div>
      <div class="viewUserTrips"
           data-v-311c04db
      >
        <h2 data-v-311c04db>
          Trips
        </h2>
        <div class="trip_preview_card wide_preview_card"
             data-v-22a19118
             data-v-311c04db
        >
          <div class="n-card n-card--bordered n-card--hoverable widePreviewCard"
               style="--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-border-radius: 3px; --n-color: #fff; --n-color-modal: #fff; --n-color-popover: #fff; --n-color-target: #18a058; --n-text-color: rgb(51, 54, 57); --n-line-height: 1.6; --n-action-color: rgb(250, 250, 252); --n-title-text-color: rgb(31, 34, 37); --n-title-font-weight: 500; --n-close-color: rgba(122, 122, 122, 1); --n-close-color-hover: rgba(89, 89, 89, 1); --n-close-color-pressed: rgba(149, 149, 149, 1); --n-border-color: rgb(239, 239, 245); --n-box-shadow: 0 1px 2px -2px rgba(0, 0, 0, .08), 0 3px 6px 0 rgba(0, 0, 0, .06), 0 5px 12px 4px rgba(0, 0, 0, .04); --n-padding-top: 19px; --n-padding-bottom: 20px; --n-padding-left: 24px; --n-font-size: 14px; --n-title-font-size: 18px; --n-close-size: 18px;"
               data-v-22a19118
          >
            <div class="n-card__content"
                 style="padding: 0px;"
                 role="none"
            >
              <a class="underline_never color_never tripLink"
                 href="/trip/view/5"
                 rel
                 target
                 data-v-018c0145
                 data-v-22a19118
              >
                <h3 class="tripTitleName"
                    data-v-22a19118
                >
                  some title
                </h3>
                <div role="separator"
                     class="n-divider n-divider--no-title"
                     style="--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-color: rgb(239, 239, 245); --n-text-color: rgb(31, 34, 37); --n-font-weight: 500;"
                     data-v-22a19118
                >
                  <div class="n-divider__line n-divider__line--left">
                  </div>
                </div>
                <div class="tripPreviewInfo"
                     data-v-22a19118
                >
                  <p data-v-22a19118>
                    <a class="underline_hover color_always tripPreviewUserProfileLink"
                       href="/user/10"
                       rel
                       target
                       data-v-018c0145
                       data-v-22a19118
                    >
                      MyAccount Test User
                    </a>
                  </p>
                  <div class="cityTags"
                       data-v-22a19118
                  >
                  </div>
                  <p data-v-22a19118>
                    Last Updated: Dec 31 2020
                  </p>
                </div>
                <div class="endOfCardPadding"
                     data-v-22a19118
                >
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`vIDUser > Get User - Not found 1`] = `
<div class="get_user"
     data-v-de4cceec
>
  <div class="head"
       data-v-de4cceec
  >
  </div>
  <h1 class="title"
      data-v-de4cceec
  >
  </h1>
  <div class="profile_info"
       data-v-de4cceec
  >
    <div class="view_user"
         data-v-311c04db
         data-v-de4cceec
    >
      <div class="userInfo narrow_content"
           data-v-311c04db
      >
        <div class="externalLink"
             data-v-311c04db
        >
        </div>
      </div>
      <div role="separator"
           class="n-divider n-divider--no-title viewUserDivider"
           style="--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-color: rgb(239, 239, 245); --n-text-color: rgb(31, 34, 37); --n-font-weight: 500;"
           data-v-311c04db
      >
        <div class="n-divider__line n-divider__line--left">
        </div>
      </div>
      <div class="viewUserTrips"
           data-v-311c04db
      >
        <h2 data-v-311c04db>
          Trips
        </h2>
      </div>
    </div>
  </div>
</div>
`;
